### DISCUSSION TITLE
fan_out_request with message_priority::high and groups

### DISCUSSION THREAD

**Discussion by NickCody** [actor-framework/actor-framework#1323] [Ideas]
It doesn't seem possible to `fan_out_request` with `message_priority::high` nor does it seem possible to implement simple scatter/gather semantics with the `groups` feature.

Fanning out and delivering to the high priority mailbox don't seem to be mutually exclusive concepts. Similarly, wondering why we can't use the group feature with the request/response semantics of `fan_out_request` ?

That said, otherwise love CAF. Thanks.

**Comment by Neverlord** [comment]
> Fanning out and delivering to the high priority mailbox don't seem to be mutually exclusive concepts.

The `fan_out_request` seems to accept a `message_priority` template parameter after the policy: https://github.com/actor-framework/actor-framework/blob/0.18.5/libcaf_core/caf/mixin/requester.hpp#L95. This should allow passing `message_priority::high`, or am I mistaken?

> wondering why we can't use the group feature with the request/response semantics

Groups are a mechanism for loose coupling. Meaning that you don't know (and cannot know) how many subscribers there are. If you don't know how many subscribers there are, a fan-out requests cannot know when it has seen all responses. That's why groups won't work.

> That said, otherwise love CAF. Thanks.

❤️

**Reply by NickCody** [reply]
Thanks for your responses and wonderful explanations!
